### ISSUE TITLE
Develop QA baseline module using Huggingface

### ISSUE THREAD

**Issue by bayoumi17m** [oscarso2000/MyCourseIndex#116]
Using the transformers module by HuggingFace, use a baseline (extractive probably) QA model and familiarize with the interface of question answering models

**Comment by bayoumi17m** [comment]
@ngjulia @yangkt @Elva, please review the code / explanation below. Some of you may have seen this already while some of you may have not. This is using the transformers library by 🤗, and will allow for using some traditional baselines that are pretty decent.
```python
import torch
from transformers import (
    BertTokenizer,
    BertForQuestionAnswering,
)
# 1
tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
model = BertForQuestionAnswering \
    .from_pretrained('bert-large-uncased-whole-word-masking-finetuned-squad')
#2
question, text = "Where do I live?", "My name is Mike and I live in Ithaca."
input_text = "[CLS] " + question + " [SEP] " + text + " [SEP]"
#3
input_ids = tokenizer.encode(input_text)
token_type_ids = [0 if i <= input_ids.index(102) else 1
    for i in range(len(input_ids))]
#4
start_scores, end_scores = model(torch.tensor([input_ids]), \
    token_type_ids=torch.tensor([token_type_ids]))
#5
all_tokens = tokenizer.convert_ids_to_tokens(input_ids)
print(' '.join(all_tokens[torch.argmax(start_scores) : torch.argmax(end_scores)+1]))
```
Below are the values  / printouts you should see
```bash
Expected answer (printed out): Berlin
# The below is not printed
Expected start: 34
Expected end: 39
Softmaxed 'score': 0.987
```

1. Loads a pre-trained tokenizer and pre-trained question answering model
2. Takes embeds the question and context into a string delimited by “[SEP]”
3. Encodes the query and generates a boolean array which marks which tokens are part of the question and which are part of the context.
4. Run the model to get two arrays with the same length as input_ids. Each element in start_scores is an softmax (normalized sigmoid) value indicating how likely it is that the token is the start of the answer. Similarly, end_scores marks the end of the answer.
5. We can extract the answer by slicing between the strongest activations of start_score and end_score.
